xla/service/cpu/onednn_contraction_rewriter.cc

```diff
@@ -993,6 +993,12 @@ class OneDnnContractionRewriteVisitor : public DfsHloRewriteVisitor {
     if (Match(instr,
               m::Copy(&copy, m::Transpose(&transpose,
                                           OneDnnMatmulInstr(&custom_call))))) {
+      auto rank = transpose->dimensions().size();
+      // oneDNN does not dispatch to optimized algorithm in these cases, hence
+      // avoid folding transposes here.
+      // TODO(intel-tf): Reevaluate this condition with future oneDNN releases.
+      if (rank >= 5 || transpose->dimensions()[rank - 1] != rank - 1)
+        return absl::OkStatus();
```

Comment:
Nit: Let's add braces even for single-line `if`s.
```suggestion
      if (rank >= 5 || transpose->dimensions()[rank - 1] != rank - 1) {
        return absl::OkStatus();
      }
```

Reply:
Done